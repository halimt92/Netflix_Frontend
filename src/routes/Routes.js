/* eslint-disable  import/no-named-as-default-member */
/* eslint-disable  import/no-named-as-default */

import React, { useEffect } from "react";
import { BrowserRouter, Switch, Route } from "react-router-dom";
import Homepage from "../container/Homepage";
import Movie from "../pages/movie-show-page/Movie";
import MovieCategory from "../pages/movie-category-page/MovieCategory";
import SearchResult from "../pages/search-result-page/SearchResult";
import Login from "../components/Login/Login";
import SignUp from "../components/SignUp/SignUp";

const Routes = () => {
  return (
    <Switch>
      <Route path="/login/" component={Login} exact />
      <Route path="/signup/" component={SignUp} exact />
      <Route path="/" component={Homepage} exact />
      <Route path="/action/" component={MovieCategory} exact />
      <Route path="/horror/" component={MovieCategory} exact />
      <Route path="/comedy/" component={MovieCategory} exact />
      <Route path="/top_rated/" component={MovieCategory} exact />
      <Route path="/search/" component={SearchResult} exact />
      <Route path="/:movie/:id" component={Movie} />
    </Switch>
  );
};

export default Routes;
