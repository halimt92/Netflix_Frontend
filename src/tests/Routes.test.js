import React from 'react';
import { shallow } from 'enzyme';
import { Route, Switch } from 'react-router';
import { BrowserRouter } from 'react-router-dom';
import Routes from '../routes/Routes';
import Navbar from '../components/navbar/Navbar';
import Footer from '../components/footer/Footer';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useDispatch: () => mockDispatch,
}));

describe('Routes', () => {
  it('should render a <Navbar /> component', () => {
    const wrapper = shallow(<Routes />);
    expect(wrapper.find(Navbar)).toHaveLength(1);
  });

  it('should render a <Footer /> component', () => {
    const wrapper = shallow(<Routes />);
    expect(wrapper.find(Footer)).toHaveLength(1);
  });

  it('should render a routes', () => {
    const wrapper = shallow(<Routes />);
    expect(wrapper.find(Route)).toHaveLength(7);
  });

  it('should render a BrowserRouter', () => {
    const wrapper = shallow(<Routes />);
    expect(wrapper.find(BrowserRouter)).toHaveLength(1);
  });

  it('should render a Switch', () => {
    const wrapper = shallow(<Routes />);
    expect(wrapper.find(Switch)).toHaveLength(1);
  });

  it('matches Render snapshot', () => {
    const wrapper = shallow(<Routes />);
    expect(wrapper).toMatchSnapshot();
  });
});
