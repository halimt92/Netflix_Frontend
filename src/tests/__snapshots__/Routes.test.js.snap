// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes matches Render snapshot 1`] = `ShallowWrapper {}`;

exports[`RowContainer matches RowContainer snapshot 1`] = `ShallowWrapper {}`;
